ci/container_index/lib/checks/basevalidation.py

```diff
@@ -0,0 +1,86 @@
+from ci.container_index.lib.utils import IndexCIMessage
+import os
+
+
+class Validator(object):
+
+    def __init__(self, validation_data, file_name):
+        self.message = IndexCIMessage(validation_data)
+        self.validation_data = validation_data
+        self.file_name = file_name
+        self.file_base_name = os.path.basename(self.file_name)
+
+    def _invalidate(self, err):
+        self.message.success = False
+        self.message.errors.append(err)
+
+    def _warn(self, warn):
+        self.message.warnings.append(warn)
+
+    def _perform_validation(self):
+        """
+        Validation logic of the validator resides here.
+        Subclass must override
+        """
+        raise NotImplementedError
+
+    def validate(self):
+        """
+        Runs the validator to validate based on provided data.
+        :return: Returns a flag to indicate success or falure
+        and an IndexCIMessage object.
+        """
+        self._perform_validation()
+        return self.message
+
+
+class BasicSchemaValidator(Validator):
+
+    def __init__(self, validation_data, file_name):
+        super(BasicSchemaValidator, self).__init__(validation_data, file_name)
+        self.field_name = ""
+
+    def _perform_validation(self):
+        if not self.validation_data.get(self.field_name):
```

Comment:
but self.field_name is empty string at line no 41 in this file, in which scenario this if statement should yield true?

Reply:
Sorted by setting a non-empty string default. That is expected to be set by the child classes. This is an abstract class and ideally, it should never run in that condition, if it does, then it looks for field which does not exist which should fail the validation.

Reply:
you should've added the same comment in code as well.